src/dex/dexalot/dexalot.ts

```diff
@@ -0,0 +1,998 @@
+    side: ClobSide,
+  ) {
+    let result = [];
+
+    for (let i = 0; i < amounts.length; i++) {
+      let amt = amounts[i];
+      if (amt === 0n) {
+        result.push(amt);
+        continue;
+      }
+
+      let left = 0,
+        right = orderbook.length;
+      let qty = BigInt(0);
+
+      while (left < right) {
+        let mid = Math.floor((left + right) / 2);
+        qty = BigInt(
+          ethers.utils
+            .parseUnits(orderbook[mid][1], quoteToken.decimals)
+            .toString(),
+        );
+        if (side === ClobSide.ASK) {
+          const price = BigInt(
+            ethers.utils
+              .parseUnits(orderbook[mid][0], baseToken.decimals)
+              .toString(),
+          );
+          qty =
+            (qty * BigInt(10 ** (baseToken.decimals * 2))) /
+            (price * BigInt(10 ** quoteToken.decimals));
+        }
+        if (qty <= amt) {
+          left = mid + 1;
+        } else {
+          right = mid;
+        }
+      }
+
+      let price,
+        amount = BigInt(0);
+      if (left === orderbook.length) {
+        price = BigInt(
+          ethers.utils
+            .parseUnits(orderbook[left - 1][0], baseToken.decimals)
+            .toString(),
+        );
+        amount = qty;
+      } else if (amounts[i] === qty) {
+        price = BigInt(
+          ethers.utils
+            .parseUnits(orderbook[left][0], baseToken.decimals)
+            .toString(),
+        );
+        amount = amounts[i];
+      } else {
+        const lPrice = BigInt(
+          ethers.utils
+            .parseUnits(orderbook[left][0], baseToken.decimals)
+            .toString(),
+        );
+        const rPrice = BigInt(
+          ethers.utils
+            .parseUnits(orderbook[left + 1][0], baseToken.decimals)
+            .toString(),
+        );
+        const lQty = qty;
+        let rQty = BigInt(
+          ethers.utils
+            .parseUnits(orderbook[left + 1][1], quoteToken.decimals)
+            .toString(),
+        );
+        if (side === ClobSide.ASK) {
+          rQty = (rQty * rPrice) / BigInt(10 ** quoteToken.decimals);
+        }
+        price = lPrice + ((rPrice - lPrice) * (amt - lQty)) / (rQty - lQty);
+        amount = amounts[i];
+      }
+
+      if (side === ClobSide.BID) {
+        result.push(
+          (amount * BigInt(10 ** (baseToken.decimals * 2))) /
+            (price * BigInt(10 ** quoteToken.decimals)),
+        );
+      } else {
+        result.push(
+          (price * amount * BigInt(10 ** quoteToken.decimals)) /
+            BigInt(10 ** (baseToken.decimals * 2)),
+        );
+      }
+    }
+    return result;
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+    transferFees?: TransferFeeParams,
+  ): Promise<null | ExchangePrices<DexalotData>> {
+    try {
+      if (await this.isRestricted()) {
+        return null;
+      }
```

Comment:
I am not sure this logic should be on dex level. We check it somewhere outside, if I am not mistaken:
https://github.com/paraswap/paraswap-api/blob/75554fd9d3eca5af1d4922998ad79504d6bda031/src/lib/hopper/chain-manager.ts#L973-L978

The same, btw, with SwaapV2. Maybe good to keep before RFQ refactor task